test/parallel/test-fs-watchfile.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const assert = require('assert');
+const common = require('../common');
+const fixtures = path.join(__dirname, '..', 'fixtures');
+
+var callbackFired = 0;
+
+// Basic usage tests.
+assert.throws(function() {
+  fs.watchFile('./some-file');
+}, /watchFile requires a listener function/);
+
+assert.throws(function() {
+  fs.watchFile('./another-file', {}, 'bad listener');
+}, /watchFile requires a listener function/);
+
+assert.throws(function() {
+  fs.watchFile(new Object(), function() {});
+}, /Path must be a string/);
+
+// Test ENOENT. Should fire once.
+const enoentFile = path.join(fixtures, 'empty', 'non-existent-file');
+fs.watchFile(enoentFile, function(curr, prev) {
```

Comment:
I had the counter in so that if the callback were fired more than once, it would error. I'm not sure if that is still possible unless it is called synchronously twice - but just in case, I'd like to err on the side of keeping the counter.